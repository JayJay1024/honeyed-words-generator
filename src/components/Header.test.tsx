import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';

import Header from './Header';

describe('component Header', () => {
    let handleStart: jest.Mock = null;
    let testRenderer: ReactTestRenderer = null;

    beforeEach(() => {
        handleStart = jest.fn();
        act(() => {
            testRenderer = create(<Header handleStart={handleStart} />);
        });
    });

    afterEach(() => {
        if (testRenderer) {
            testRenderer.unmount();
            testRenderer = null;
        }
    });

    test('Header should match snapshot', () => {
        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('there should be some title words', () => {
        const testInstance = testRenderer.root;
        expect(testInstance.findAllByProps({className: "pinyin"}).length).toEqual(4);
        expect(testInstance.findAllByProps({className: "pinyin"})[0].parent.children).toContain("甜");
        expect(testInstance.findAllByProps({className: "pinyin"})[1].parent.children).toContain("言");
        expect(testInstance.findAllByProps({className: "pinyin"})[2].parent.children).toContain("蜜");
        expect(testInstance.findAllByProps({className: "pinyin"})[3].parent.children).toContain("语");
    });

    test('there should be a button to start', () => {
        const testInstance = testRenderer.root;
        expect(testInstance.findAllByType("button")).toHaveLength(1);
        expect(testInstance.findByType("button").children).toContain("开始");
    })
});
