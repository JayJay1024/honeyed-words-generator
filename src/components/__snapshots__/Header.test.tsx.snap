// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Header Header should match snapshot 1`] = `
<header
  className="sc-AxiKw cBIvkF"
>
  <img
    alt="logo"
    className="logo"
    src={Object {}}
  />
  <div
    className="title"
  >
    <span
      className="sc-AxjAm sc-AxhCb fVqQdo"
      style={
        Object {
          "animationDelay": "0s",
        }
      }
    >
      甜
      <em
        className="pinyin"
      >
        <i
          className="w"
          style={
            Object {
              "animationDelay": "1.2s",
            }
          }
        >
          tǔ
        </i>
      </em>
    </span>
    <span
      className="sc-AxjAm sc-AxhCb fVqQdo"
      style={
        Object {
          "animationDelay": "0.3s",
        }
      }
    >
      言
      <em
        className="pinyin"
      >
        <i
          className="w"
          style={
            Object {
              "animationDelay": "1.5s",
            }
          }
        >
          wèi
        </i>
      </em>
    </span>
    <span
      className="sc-AxjAm sc-AxhCb fVqQdo"
      style={
        Object {
          "animationDelay": "0.6s",
        }
      }
    >
      蜜
      <em
        className="pinyin"
      >
        <i
          className="w"
          style={
            Object {
              "animationDelay": "1.7999999999999998s",
            }
          }
        >
          qíng
        </i>
      </em>
    </span>
    <span
      className="sc-AxjAm sc-AxhCb fVqQdo"
      style={
        Object {
          "animationDelay": "0.8999999999999999s",
        }
      }
    >
      语
      <em
        className="pinyin"
      >
        <i
          className="w"
          style={
            Object {
              "animationDelay": "2.0999999999999996s",
            }
          }
        >
          huà
        </i>
      </em>
    </span>
  </div>
  <button
    className="sc-AxirZ kYecUW"
    onClick={[MockFunction]}
  >
    开始
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "0s",
        }
      }
    />
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "0.3s",
        }
      }
    />
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "0.6s",
        }
      }
    />
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "0.8999999999999999s",
        }
      }
    />
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "1.2s",
        }
      }
    />
    <img
      alt="heart"
      className="heart"
      src={Object {}}
      style={
        Object {
          "animationDelay": "1.5s",
        }
      }
    />
  </button>
</header>
`;
